Mobile/app/src/main/java/com/example/squirrel/AuthorizationActivity.kt

```diff
@@ -45,61 +47,23 @@ class AuthorizationActivity : AppCompatActivity()  {
             return
         }
         val context = this
-        val jsonBody = JSONObject()
-        jsonBody.put("email", binding.loginPrompt.text.toString())
-        jsonBody.put("password", binding.passwordPrompt.text.toString())
-        val requestBody = jsonBody.toString()
-        val stringRequest = StringRequest(
-            Request.Method.POST,
-            "http://geneirodan.zapto.org:23451/api/account/authenticate?callbackurl=geneirodan.zapto.org",
-            {
+        lifecycleScope.launch{
+            val response: HttpResponse = client.submitForm(
+                url = "$protocol://$domain:$port/api/account/login",
+                formParameters = Parameters.build {
+                    append("email", binding.loginPrompt.text.toString())
+                    append("password", binding.passwordPrompt.text.toString())
+                }
+            )
+            if(response.status == HttpStatusCode.OK) {
                 Toast.makeText(context, "Welcome!", Toast.LENGTH_LONG).show()
                 startActivity(intent)
-            },
-            { error ->
-                val response = error.networkResponse
-                if (response != null) {
-                    try {
-                        val res = String(response.data, Charsets.UTF_8)
-                        val arr = JSONArray(res).toList()
-                        for(x in arr)
-                            Toast.makeText(context, x.toString(), Toast.LENGTH_SHORT).show()
-                    } catch (e1 : UnsupportedEncodingException) {
-                        e1.printStackTrace()
-                    } catch (e2 : JSONException) {
-                        e2.printStackTrace()
-                    }
-                }
-            }, requestBody)
-        Program.getRequestQueue()!!.add(stringRequest)
-    }
-    @Throws(JSONException::class)
-    fun JSONObject.toMap(): Map<String, Any> {
-        val map = mutableMapOf<String, Any>()
-        val keysItr: Iterator<String> = this.keys()
-        while (keysItr.hasNext()) {
-            val key = keysItr.next()
-            var value: Any = this.get(key)
-            when (value) {
-                is JSONArray -> value = value.toList()
-                is JSONObject -> value = value.toMap()
             }
-            map[key] = value
-        }
-        return map
-    }
-
-    @Throws(JSONException::class)
-    fun JSONArray.toList(): List<Any> {
-        val list = mutableListOf<Any>()
-        for (i in 0 until this.length()) {
-            var value: Any = this[i]
-            when (value) {
-                is JSONArray -> value = value.toList()
-                is JSONObject -> value = value.toMap()
+            else {
+                val arr = response.body<List<String>>()
```

Comment:
As far as I know even though API tries to send an array of strings-errors, it anyway sends the only string. Correct me if I'm wrong

Reply:
Yes, but we use ContentNegotiation feature of ktor. It parses JSON to chosen type.